plugins/woocommerce/src/Blocks/BlockTypes/MiniCart.php

```diff
@@ -282,7 +314,67 @@ public function print_lazy_load_scripts() {
 			);
 		}
 
-		$data                          = rawurlencode( wp_json_encode( $this->scripts_to_lazy_load ) );
+		$this->add_inline_script_data();
+	}
+
+	/**
+	 * Load the frontend dependencies cache when the cart has items.
+	 *
+	 * @return string|null String of JSON data.
+	 */
+	protected function load_frontend_dependencies_cache() {
+		if ( wp_is_development_mode( 'plugin' ) ) {
```

Comment:
I use `define( 'WP_ENVIRONMENT_TYPE', 'local' );` which I can't remember what for now but may be something to consider adding here

Reply:
I don't think we should be the one defining `WP_ENVIRONMENT_TYPE`, it comes from WP-config

Reply:
Ah yeah that's what `wp_is_development_mode` is doing here + other checks and filters

https://developer.wordpress.org/reference/functions/wp_get_development_mode/

Reply:
Doesn't look like it checks `WP_ENVIRONMENT_TYPE` only `WP_DEVELOPMENT_MODE`. They seem like they do similar things so using that function is probably good enough.

Reply:
ah good point, I didn’t know there exists 2, sorry.

Reply:
This is relatively new ([available since WP 6.3](https://make.wordpress.org/core/2023/07/14/configuring-development-mode-in-6-3/)) and differs slightly from the previous `WP_ENVIRONMENT_TYPE`, `WP_DEBUG`, and `SCRIPT_DEBUG`.

The others relate to other aspects of the environment, while `wp_is_development_mode` is commonly used for asset caching in WP core.